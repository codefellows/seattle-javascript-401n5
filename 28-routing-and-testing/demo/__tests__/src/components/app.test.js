import React from 'react';
import renderer from 'react-test-renderer';

import App from '../../../src/components/app.js';

describe('<App/> (Enzyme Test)', () => {
  it('is alive at application start', () => {
    let app = shallow(<App/>);
    expect(app.find('main').exists()).toBeTruthy();
    expect(app.find('Header').exists()).toBeTruthy();
  });
});

describe('<App/> (Snapshot Test)', () => {
  it('renders right', () => {
    const component = renderer.create(
      <App />
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();

  });
});
