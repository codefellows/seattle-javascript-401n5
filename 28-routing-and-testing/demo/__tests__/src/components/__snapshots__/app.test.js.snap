// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> (Snapshot Test) renders right 1`] = `
Array [
  <header>
    <h2>
      Our Basic App
    </h2>
  </header>,
  <main>
    <h2>
      Hello World
    </h2>
  </main>,
  <footer>
    <h3>
      You hit rock bottom
    </h3>
  </footer>,
]
`;
